src/jit/importer.cpp

```diff
@@ -1407,6 +1382,28 @@ GenTree* Compiler::impAssignStructPtr(GenTree*             destAddr,
         asgType     = impNormStructType(structHnd);
         src->gtType = asgType;
     }
+    if ((dest == nullptr) && (destAddr->OperGet() == GT_ADDR))
+    {
+        GenTree* destNode = destAddr->gtGetOp1();
+        // If the actual destination is a local, or already a block node, or is a node that
+        // will be morphed, don't insert an OBJ(ADDR) if it already has the right type.
+        if ((destNode->gtOper == GT_INDEX) || destNode->OperIsBlk() || (destNode->OperGet() == GT_LCL_VAR))
```

Comment:
This can be expressed better as:

`if (destNode->OperIs(GT_LCL_VAR, GT_INDEX) ||  destNode->OperIsBlk())`

Also the comment above should be updated to:

` // If the actual destination is a local, a GT_INDEX or  a block node,  ...`